spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,9 +127,47 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		final List<FileTailingIdleEvent> events = new ArrayList<>();
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.warn(event);
+				events.add(tailEvent);
+			}
+		});
+		File file = new File(testDir, "foo");
+		file.delete();
+		FileOutputStream foo = new FileOutputStream(file);
+		for (int i = 0; i < 50; i++) {
+			foo.write(("hello" + i + "\n").getBytes());
+		}
+		foo.flush();
+		foo.close();
+		adapter.setFile(file);
+		QueueChannel outputChannel = new QueueChannel();
+		adapter.setOutputChannel(outputChannel);
+		adapter.setIdleEventInterval(1000);
+		adapter.afterPropertiesSet();
+		adapter.start();
+		Thread.sleep(4000);
```

Comment:
The `Thread.sleep()` is always evil for test-case.
First of all you may not need to sleep so much for condition to appear.
But on the other hand our long-term experience demonstrates that we can't trust `Thread.sleep()` because there is no guarantee that condition happens during that time on slow or busy machine like our CI server.

To to reach a goal you should use `CountDownLatch` and wait for its result here instead of `Thread.sleep()` .
You than `countDown()` it in that `event ->` Lamba.
In the end I'd really would like to see assert for the `FileTailingIdleEvent` type.

Hope that is clear.